codec/src/config.rs

```diff
@@ -73,6 +76,64 @@ impl<T: Copy + PartialOrd> From<core::ops::RangeFull> for RangeCfg<T> {
     }
 }
 
+macro_rules! impl_from_nonzero {
+    ($nz_ty:ty, $ty:ty) => {
+        impl From<RangeCfg<$nz_ty>> for RangeCfg<$ty> {
+            fn from(value: RangeCfg<$nz_ty>) -> Self {
+                let start = match value.start {
+                    Bound::Included(nz) => Bound::Included(nz.get()),
+                    Bound::Excluded(nz) => Bound::Excluded(nz.get()),
+                    Bound::Unbounded => Bound::Unbounded,
+                };
+                let end = match value.end {
+                    Bound::Included(nz) => Bound::Included(nz.get()),
+                    Bound::Excluded(nz) => Bound::Excluded(nz.get()),
+                    Bound::Unbounded => Bound::Unbounded,
+                };
+                RangeCfg { start, end }
+            }
+        }
+    };
+}
+
+impl_from_nonzero!(NonZeroUsize, usize);
+impl_from_nonzero!(NonZeroU8, u8);
+impl_from_nonzero!(NonZeroU16, u16);
+impl_from_nonzero!(NonZeroU32, u32);
+impl_from_nonzero!(NonZeroU64, u64);
+
+macro_rules! impl_from_nonzero_to_usize {
```

Comment:
This ends up being a very helpful tool when going from quorums/polys to vec config.